src/objects/api/validators.py

```diff
@@ -68,32 +68,54 @@ def __call__(self, new_value, serializer_field):
             raise serializers.ValidationError(self.message, code=self.code)
 
 
+def validate_data_attr_value_part(value_part: str, code: str):
+    try:
+        variable, operator, val = value_part.rsplit("__", 2)
+    except ValueError:
+        message = _(
+            "Filter expression '%(value_part)s' doesn't have the shape 'key__operator__value'"
+        ) % {"value_part": value_part}
+        raise serializers.ValidationError(message, code=code)
+
+    if operator not in Operators.values:
+        message = _("Comparison operator `%(operator)s` is unknown") % {
+            "operator": operator
+        }
+        raise serializers.ValidationError(message, code=code)
+
+    if operator not in (
+        Operators.exact,
+        Operators.icontains,
+        Operators.in_list,
+    ) and isinstance(string_to_value(val), str):
+        message = _(
+            "Operator `%(operator)s` supports only dates and/or numeric values"
+        ) % {"operator": operator}
+        raise serializers.ValidationError(message, code=code)
+
+
 def validate_data_attrs(value: str):
+    # todo remove when 'data_attrs' filter is removed
     code = "invalid-data-attrs-query"
     parts = value.split(",")
 
     for value_part in parts:
-        try:
-            variable, operator, val = value_part.rsplit("__", 2)
-        except ValueError as exc:
-            raise serializers.ValidationError(exc.args[0], code=code) from exc
-
-        if operator not in Operators.values:
-            message = _("Comparison operator `%(operator)s` is unknown") % {
-                "operator": operator
-            }
-            raise serializers.ValidationError(message, code=code)
+        validate_data_attr_value_part(value_part, code)
 
-        if operator not in (
-            Operators.exact,
-            Operators.icontains,
-            Operators.in_list,
-        ) and isinstance(string_to_value(val), str):
+
+def validate_data_attr(value: list):
+    code = "invalid-data-attr-query"
+
+    for value_part in value:
+        # check that comma can be only in the value part
+        if "," in value_part.rsplit("__", 1)[0]:
             message = _(
-                "Operator `%(operator)s` supports only dates and/or numeric values"
-            ) % {"operator": operator}
+                "Filter expression '%(value_part)s' doesn't have the shape 'key__operator__value'"
+            ) % {"value_part": value_part}
```

Comment:
Since this error specifically occurs when there are commas outside the value part, maybe it would be good to explicitly mention that that isn't allowed in the error, something like: `Filter expression '%(value_part)s' must have the shape 'key__operator__value', commas can only be present in the 'value'`